sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/InsertAdaptiveSparkPlan.scala

```diff
@@ -102,6 +103,8 @@ case class InsertAdaptiveSparkPlan(
         // update correctly.
         case i: InMemoryTableScanExec
```

Comment:
shall we remove this case? If `CAN_CHANGE_CACHED_PLAN_OUTPUT_PARTITIONING` is false, `InMemoryTableScanExec` will not be AQEed.

Reply:
They can have different configurations, it is a cornor case though, 
1. people first tune `CAN_CHANGE_CACHED_PLAN_OUTPUT_PARTITIONING` on and cache a plan
2. people tune `CAN_CHANGE_CACHED_PLAN_OUTPUT_PARTITIONING` off and run that plan

then if we remove this pattern, then it won't apply AQE for the execued plan..